include/sharg/detail/format_base.hpp

```diff
@@ -213,65 +214,61 @@ class format_help_base : public format_base
+    void add_flag(bool & SHARG_DOXYGEN_ONLY(value), config_type const & config)
     {
-        std::string id = prep_id_for_help(short_id, long_id);
+        std::string id = prep_id_for_help(config.short_id, config.long_id);
         store_help_page_element(
-            [this, id, desc]()
+            [this, id, config]()
             {
-                derived_t().print_list_item(id, desc);
+                derived_t().print_list_item(id, config.description);
             },
-            spec);
+            config.advanced,
+            config.hidden);
     }
 
     /*!\brief Adds a sharg::print_list_item call to be evaluated later on.
      * \copydetails sharg::parser::add_positional_option
      */
-    template <typename option_type, typename validator_type>
-    void add_positional_option(option_type & value, std::string const & desc, validator_type & option_validator)
+    template <typename option_type, typename config_type>
+    void add_positional_option(option_type & value, config_type const & config)
     {
-        std::string msg = option_validator.get_help_page_message();
+        std::string msg = config.validator.get_help_page_message();
 
-        // clang-format off
-        positional_option_calls.push_back([this, &value, desc, msg] ()
-        {
-            ++positional_option_count;
-            derived_t().print_list_item(detail::to_string("\\fBARGUMENT-", positional_option_count, "\\fP ",
-                                                          option_type_and_list_info(value)),
-                                        desc +
-                                        // a list at the end may be empty and thus have a default value
-                                        ((detail::is_container_option<option_type>)
-                                            ? detail::to_string(" Default: ", value, ". ")
-                                            : std::string{" "}) +
-                                        msg);
-        });
+        positional_option_calls.push_back(
+            [this, &value, config, msg]()
+            {
+                ++positional_option_count;
+                derived_t().print_list_item(detail::to_string("\\fBARGUMENT-",
+                                                              positional_option_count,
+                                                              "\\fP ",
+                                                              option_type_and_list_info(value)),
+                                            config.description +
+                                                // a list at the end may be empty and thus have a default value
+                                                ((detail::is_container_option<option_type>)
+                                                     ? detail::to_string(" Default: ", value, ". ")
+                                                     : std::string{" "})
+                                                + msg);
```

Comment:
```suggestion
                                                + config.validator.get_help_page_message());
```

Would this work? Then you don't need the `msg`.